plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/procedure/FlushHiveMetastoreCacheProcedure.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.procedure;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.plugin.hive.metastore.cache.CachingHiveMetastore;
+import io.trino.spi.procedure.Procedure;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Optional;
+
+import static io.trino.spi.block.MethodHandleUtil.methodHandle;
+import static java.util.Objects.requireNonNull;
+
+public class FlushHiveMetastoreCacheProcedure
+        implements Provider<Procedure>
+{
+    private static final MethodHandle FLUSH_HIVE_METASTORE_CACHE = methodHandle(
+            FlushHiveMetastoreCacheProcedure.class,
+            "flushMetadataCache");
+
+    private final Optional<CachingHiveMetastore> cachingHiveMetastore;
+
+    @Inject
+    public FlushHiveMetastoreCacheProcedure(Optional<CachingHiveMetastore> cachingHiveMetastore)
+    {
+        this.cachingHiveMetastore = requireNonNull(cachingHiveMetastore, "cachingHiveMetastore is null");
+    }
+
+    @Override
+    public Procedure get()
+    {
+        return new Procedure(
+                "system",
+                "flush_metadata_cache",
+                ImmutableList.of(),
+                FLUSH_HIVE_METASTORE_CACHE.bindTo(this));
+    }
+
+    public void flushMetadataCache()
+    {
+        cachingHiveMetastore.ifPresent(cache -> cache.flushCache());
```

Comment:
Does it make sense to invoke the procedure when `cachingHiveMetastore` is not present?
Looks like we could throw here instead.

Reply:
Hmm my optics when doing that where that in this case NOP is pretty ok. 

I assumed that people may use it in larger chunks of SQL just in case if cache is enabled.

So if cache is disabled.
There is no issue and flush statement is not needed. Even if it's there it should be NOP.

If it's enabled:
We might have issues with outdated cache and cache flush will flush it.

Reply:
OTOH, i don't want people to blindly invoke `flush()` thinking they are doing something that _might_ fix a problem they are observing. 

@losipiuk another POV?

Reply:
"proof of value" ? ;)

I think NOOP is OK. It is less traction. They should quickly learn that it does not fix their problem if they do not have cache enabled.

Reply:
point of view